packages/cli/src/ActiveExecutions.ts

```diff
@@ -64,10 +67,14 @@ export class ActiveExecutions {
 			if (executionId === undefined) {
 				throw new ApplicationError('There was an issue assigning an execution id to the execution');
 			}
+
+			await this.concurrencyControl.check({ mode, executionId });
```

Comment:
can we rename `.check` to something else that more clearly indicates that the execution will wait here until there is available capacity?

Reply:
I sympathize! But I cannot think of a reasonably short method name that is accurate enough. The execution will wait here only if there is no spare capacity, else it goes through, like a checkpoint. `blockOrLetThrough` sounds clunky and still not clear enough. 

Since on hover we have the docline `Block or let through an execution based on concurrency capacity`, I expect this is one of those cases where the symbol name can be short as long as properly explained. Let me know if you have a better idea.

Reply:
I'll rename - my concern was making it clear we _may or may not_ be blocking here, but I don't feel strongly about this.